src/commands/lambda/functions/instrument.ts

```diff
@@ -267,47 +257,72 @@ export const calculateUpdateRequest = async (
     }
   }
 
-  if (runtime === DOTNET_RUNTIME) {
-    needsUpdate = true
-    newEnvVars[ENABLE_PROFILING_ENV_VAR] = CORECLR_ENABLE_PROFILING
-    newEnvVars[PROFILER_ENV_VAR] = CORECLR_PROFILER
-    newEnvVars[PROFILER_PATH_ENV_VAR] = CORECLR_PROFILER_PATH
-    newEnvVars[DOTNET_TRACER_HOME_ENV_VAR] = DD_DOTNET_TRACER_HOME
-  }
-
-  updateRequest.Environment = {
-    Variables: newEnvVars,
-  }
-
   let layerARNs = getLayers(config)
   const originalLayerARNs = layerARNs
   let needsLayerUpdate = false
+  let layerOrTraceVersion: number | undefined
   if (isLayerRuntime(runtime)) {
     const lambdaLibraryLayerArn = getLayerArn(config, config.Runtime as LayerKey, region, settings)
     const lambdaLibraryLayerName = LAYER_LOOKUP[runtime as LayerKey]
     let fullLambdaLibraryLayerARN: string | undefined
     if (settings.layerVersion !== undefined || settings.interactive) {
-      let layerVersion = settings.layerVersion
+      layerOrTraceVersion = settings.layerVersion
       if (settings.interactive && !settings.layerVersion) {
-        layerVersion = await findLatestLayerVersion(config.Runtime as LayerKey, region)
+        layerOrTraceVersion = await findLatestLayerVersion(config.Runtime as LayerKey, region)
       }
-      fullLambdaLibraryLayerARN = `${lambdaLibraryLayerArn}:${layerVersion}`
+      fullLambdaLibraryLayerARN = `${lambdaLibraryLayerArn}:${layerOrTraceVersion}`
     }
     layerARNs = addLayerArn(fullLambdaLibraryLayerARN, lambdaLibraryLayerName, layerARNs)
   }
 
   const lambdaExtensionLayerArn = getLayerArn(config, EXTENSION_LAYER_KEY as LayerKey, region, settings)
   let fullExtensionLayerARN: string | undefined
+  let extensionVersion: number | undefined
   if (settings.extensionVersion !== undefined || settings.interactive) {
-    let extensionVersion = settings.extensionVersion
+    extensionVersion = settings.extensionVersion
     if (settings.interactive && !settings.extensionVersion) {
       extensionVersion = await findLatestLayerVersion(EXTENSION_LAYER_KEY as LayerKey, region)
     }
     fullExtensionLayerARN = `${lambdaExtensionLayerArn}:${extensionVersion}`
   }
-
   layerARNs = addLayerArn(fullExtensionLayerARN, DD_LAMBDA_EXTENSION_LAYER_NAME, layerARNs)
 
+  // Special handling for .NET and Java to support universal instrumentation
+  if (runtimeType === RuntimeType.DOTNET || runtimeType === RuntimeType.JAVA) {
+    if (layerOrTraceVersion && isExtensionSupportUniversalInstrumentation(runtimeType, extensionVersion)) {
```

Comment:
How about `extensionSupportsUniversalInstrumentation` or 
`isExtensionCompatibleWithUniversalInstrumentation`. 

isExtensionSupportUniversalInstrumentation sounds grammatically incorrect to me.